parser/prism/Translator.cc

```diff
@@ -2232,7 +2232,10 @@ ast::ExpressionPtr Translator::desugar(pm_node_t *node) {
             }
 
             // Wrap in an InsSeq with the predicate assignment (if there is a predicate)
-            auto predicate = desugar(caseMatchNode->predicate);
+            auto predicate = desugarNullable(caseMatchNode->predicate);
+            if (ast::isa_tree<ast::EmptyTree>(predicate)) {
+                return resultExpr;
+            }
```

Comment:
What do you think about checking for null directly:

```suggestion
            if (caseMatchNode->predicate == nullptr) {
                 return resultExpr;
            }
            
            auto predicate = desugar(caseMatchNode->predicate);
```

Reply:
Much nicer, I don't know what I was thinking. Have fixed up now.